gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -306,6 +303,96 @@ public String apply(KafkaTopic topic) {
     }
   }
 
+  public List<WorkUnit> getWorkunitsForFilteredPartitions(SourceState state, Map<String, List<String>> filteredTopicPartitionMap, int minContainer) {
```

Comment:
why min container is needed here and how will we determine it?

Reply:
Give an option to specify a numContainers that used by KafkaTopicGroupingWorkUnitPacker#pack . How it being used is determined by the implementation of concrete KafkaWorkUnitPacker class.

Reply:
My next PR will be utilizing this minContainer in KafkaTopicGroupingWorkUnitPacker, which is not being respected in the current implementation.